app/views/grant_submissions/submissions/index.html.haml

```diff
@@ -43,22 +47,59 @@
           %th
             = sort_link(@q, :state, 'State')
           %th
-            = sort_link(@q, :updated_at, 'Last Updated')
-          %th
+            = sort_link(@q, :updated_at, 'Updated')
+          - if current_user.get_role_by_grant(grant: @grant)
+            %th
+              Reviews
+            %th
+              %span{ title: 'Average of scored reviews' }
+                Overall Impact
+            %th
+              %span{ title: 'Average of scored Criteria' }
+                Composite
+
       %tbody
         - @submissions.each do |submission|
-          %tr
+          - submission_reviews = submission.reviews.to_a
+          - has_completed_reviews = submission_reviews.any?{ |review| review.overall_impact_score.present? }
+
+          %tr{data: { submission: submission.id } }
             %td
               = link_to sortable_full_name(submission.applicant), grant_submission_path(@grant, submission)
             %td
               = link_to submission.title, grant_submission_path(@grant, submission)
             %td
-              = submission.state.titleize
-            %td
-              = submission.updated_at
+              %div.status{data: {status: submission.id} }
+                = submission.state.titleize
+              - if current_user.get_role_by_grant(grant: @grant) != 'viewer' && current_user != submission.applicant
```

Comment:
why not reuse these existing partials conditionally by role/current_user?app/views/grant_submissions/submissions/_admin_links.html.haml
app/views/grant_submissions/submissions/_editor_links.html.haml
if there's a slight different than the existing usage, maybe try passing options to it?